### ISSUE TITLE
New Rule Idea: Enums not marked with `FlagsAttribute` should not use `Enum.HasFlag` method 

### ISSUE THREAD

**Issue by sebastien-marichal** [SonarSource/sonar-dotnet#8197]
From this [community post](https://community.sonarsource.com/t/rule-to-raise-a-bug-for-hasflag-usage-on-non-flag-enums/102389).

As stated in [Enum.HasFlag documentation](https://learn.microsoft.com/en-us/dotnet/api/system.enum.hasflag):
> The [HasFlag](https://learn.microsoft.com/en-us/dotnet/api/system.enum.hasflag?view=net-7.0) method is designed to be used with enumeration types that are marked with the [FlagsAttribute](https://learn.microsoft.com/en-us/dotnet/api/system.flagsattribute?view=net-7.0) attribute [...]

#### Noncompliant  code

```cs
enum MyEnum
{
    None,
    First,
    Second,
    Third
}

Console.WriteLine(MyEnum.First.HasFlag(MyEnum.Second)); // Noncompliant
```

#### Compliant code

```cs
[Flags]
enum MyEnum
{
    None = 0,
    First = 1,
    Second = 2,
    Third = 4
}

Console.WriteLine(MyEnum.First.HasFlag(MyEnum.Second)); // Compliant
```

**Comment by rjgotten** [comment]
This already exists as CA2248, a Roslyn analyzer written by MS itself.
https://github.com/dotnet/roslyn-analyzers/issues/3528

**Comment by sebastien-marichal** [comment]
Hello @rjgotten,

After taking a look it seems these changes have been reverted afterward: https://github.com/dotnet/roslyn-analyzers/pull/4575 

As of today, it seems CA2248 does not report this case.

As a note for implementation, it seems Roslyn reverted those changes due to a high rate of FP.
Take a look at https://github.com/dotnet/roslyn-analyzers/issues/4432 for potential FP.

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1660](https://sonarsource.atlassian.net/browse/NET-1660)

[NET-1660]: https://sonarsource.atlassian.net/browse/NET-1660?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
